### DISCUSSION TITLE
Integration testing guide

### DISCUSSION THREAD

**Discussion by molebox** [prisma/prisma#7088] [Q&A]
This is a space for community members to provide feedback on and participate in discussions surrounding our [integration testing guide](https://www.prisma.io/docs/guides/testing/integration-testing).

Bear in mind that this is a first iteration and as such doesnt cover every known use case. Moving forward we would like to cover as much ground as possible with these guides. 

Some handy stuff we would like to know:
- Do the examples given in the guide cover your testing use cases?
- What other scenarios do you have? How do _you_ cover those scenarios if different to how we have done it?
- If you have a different scenario, an example repo which shows how your setup runs would be awesome

And of course any other feedback you might have is most welcome 😀

**Comment by nahtnam** [comment]
I suggest not using the 'default' schema and instead just generate a UUID. The reason is if someone accidentally forgets to swap the DATABASE_URL, they will drop their actual DB. This is how I do it: https://github.com/prisma/prisma/discussions/2083#discussioncomment-714248

**Reply by molebox** [reply]
This looks really neat, thanks @nahtnam. I'll setup a repo and checkout how it works. If all is good i can modify our docs. Appreciate your feedback 🙌

**Comment by jasonlimantoro** [comment]
For `MySQL` which does not support `TRUNCATED [TABLE_NAME] CASCADE`, you need to do the following for truncating tables:

```ts
import { prisma } from '../../src/database/prisma';
import { PrismaPromise } from '@prisma/client';

async function truncateTables() {
	const transactions: PrismaPromise<any>[] = [];
	transactions.push(prisma.$executeRaw(`SET FOREIGN_KEY_CHECKS = 0;`));

	const tables = await prisma.$queryRaw<
		{ TABLE_NAME: string }[]
	>`SELECT TABLE_NAME from information_schema.TABLES WHERE TABLE_SCHEMA = "test_db";`;
	for (const { TABLE_NAME } of tables) {
		if (TABLE_NAME !== "_prisma_migrations") {
			transactions.push(prisma.$queryRaw(`TRUNCATE TABLE ${TABLE_NAME};`));
		}
	}

	transactions.push(prisma.$executeRaw(`SET FOREIGN_KEY_CHECKS = 1;`));
	await prisma.$transaction(transactions);
};
```

Reference: 
- https://www.prisma.io/docs/concepts/components/prisma-client/crud#delete-all-records-from-all-tables

**Reply by molebox** [reply]
Thanks for the feedback @jasonlimantoro , indeed its a good idea to add a mysql example. Ill create an [issue](https://github.com/prisma/docs/issues/1704) for adding this example. 🙌

**Comment by ulysset** [comment]
Hi, thanks for the docs, they are really useful :)

> The guide uses a singleton pattern for Prisma Client setup. Refer to the singleton docs for a walk through of how to set that up.

In the docs you mention that for integration testing you are using the `singleton` approach. But if we use this approach the Prisma Client will be mocked so we won't be able to CRUD the DB. Shouldn't we use the dependency injection approach? https://www.prisma.io/docs/guides/testing/unit-testing#dependency-injection to only mock the prisma client in unit tests and leave it as is (unmocked) when integration testing?

**Reply by jasonlimantoro** [reply]
Yeah that was confusing me for some time. Perhaps what they mean is to setup the singleton prisma instance only (i.e., Step 1 only).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
